src/julia.h

```diff
@@ -877,6 +876,15 @@ static inline uint32_t jl_fielddesc_size(int8_t fielddesc_type)
 #define jl_is_cpointer(v)    jl_is_cpointer_type(jl_typeof(v))
 #define jl_is_pointer(v)     jl_is_cpointer_type(jl_typeof(v))
 
+JL_DLLEXPORT int jl_subtype(jl_value_t *a, jl_value_t *b);
+
+STATIC_INLINE int jl_is_type(jl_value_t *v)
+{
+    jl_value_t *t = jl_typeof(v);
+    return (t == (jl_value_t*)jl_datatype_type || t == (jl_value_t*)jl_uniontype_type ||
+            t == (jl_value_t*)jl_unionall_type || v == jl_bottom_type);
```

Comment:
in the spirit of DRY, we could define this as `#define jl_is_type(v) (jl_is_kind(jl_typeof(v)))`